test/gtest/pdist.cpp

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "pdist.hpp"
+using float16 = half_float::half;
+
+// BACKWARD TEST
+using GPU_Pdist_bwd_FP32  = PdistTestBackward<float>;
+using GPU_Pdist_bwd_FP16  = PdistTestBackward<float16>;
+using GPU_Pdist_bwd_BFP16 = PdistTestBackward<bfloat16>;
+
+TEST_P(GPU_Pdist_bwd_FP32, PdistTestBackward)
+{
+    RunTest();
+    Verify();
+};
+
+TEST_P(GPU_Pdist_bwd_FP16, PdistTestBackward)
+{
+    RunTest();
+    Verify();
+};
+
+TEST_P(GPU_Pdist_bwd_BFP16, PdistTestBackward)
+{
+    RunTest();
+    Verify();
+};
+
+INSTANTIATE_TEST_SUITE_P(Smoke, GPU_Pdist_bwd_FP32, testing::ValuesIn(PdistTestConfigs()));
+INSTANTIATE_TEST_SUITE_P(Smoke, GPU_Pdist_bwd_FP16, testing::ValuesIn(PdistFp16TestConfigs()));
+INSTANTIATE_TEST_SUITE_P(Smoke, GPU_Pdist_bwd_BFP16, testing::ValuesIn(PdistTestConfigs()));
+
+INSTANTIATE_TEST_SUITE_P(Full, GPU_Pdist_bwd_FP32, testing::ValuesIn(PdistTestConfigs()));
+INSTANTIATE_TEST_SUITE_P(Full, GPU_Pdist_bwd_FP16, testing::ValuesIn(PdistFp16TestConfigs()));
+INSTANTIATE_TEST_SUITE_P(Full, GPU_Pdist_bwd_BFP16, testing::ValuesIn(PdistTestConfigs()));
+
+INSTANTIATE_TEST_SUITE_P(Perf, GPU_Pdist_bwd_FP32, testing::ValuesIn(PdistTestConfigs()));
+INSTANTIATE_TEST_SUITE_P(Perf, GPU_Pdist_bwd_FP16, testing::ValuesIn(PdistFp16TestConfigs()));
+INSTANTIATE_TEST_SUITE_P(Perf, GPU_Pdist_bwd_BFP16, testing::ValuesIn(PdistTestConfigs()));
```

Comment:
I fixed this.

My additional question: Is there any references that say we only need to do test_type of `Full`. I included all types of test because I refered to the op `prelu`.

Reply:
you have only one test set, so it is of TEST_TYPE = Full. Check https://github.com/ROCm/MIOpen/wiki/GTest-development for more in formation